epochStart/bootstrap/process.go

```diff
@@ -467,11 +444,23 @@ func (e *epochStartBootstrap) computeIfCurrentEpochIsSaved() bool {
 }
 
 func (e *epochStartBootstrap) prepareComponentsToSyncFromNetwork() error {
-	err := e.createTriesComponentsForShardId(core.MetachainShardId)
+	e.closeTrieComponents()
```

Comment:
It is ok, because if they are nil here, nothing will happen, and if they are not nil, they will be closed. After this close is called, these components will be reinitialized on L448, L461, L462, and these will be closed on the defer from L284.